src/surface.rs

```diff
@@ -150,6 +151,19 @@ impl Surface {
         }
     }
 
+    pub fn get_device(&self) -> Option<Device> {
+        unsafe {
+            let device = ffi::cairo_surface_get_device(self.to_raw_none());
+            if device.is_null() {
+                None
+            } else {
+                Some(Device::from_raw_none(ffi::cairo_surface_get_device(
+                    self.to_raw_none(),
+                )))
```

Comment:
Uhm... How about `Some(Device::from_raw_none(device))` instead?

Reply:
Outch indeed! I went way too quickly in here...